common/repositories/inventory_snapshots_repository.h

```diff
@@ -68,6 +95,227 @@ class InventorySnapshotsRepository: public BaseInventorySnapshotsRepository {
+	{
+		if (!character_id || !timestamp) {
+			return false;
+		}
+
+		const std::string &query = fmt::format(
+			"SELECT * FROM `inventory_snapshots` WHERE `character_id` = '{}' "
+			"AND `time_index` = '{}' LIMIT 1",
+			character_id,
+			timestamp
+		);
+		auto results = db.QueryDatabase(query);
+
+		if (!results.Success() || results.RowCount() == 0) {
+			return false;
+		}
+
+		return true;
+	}
+
+	static void ParseCharacterInvSnapshot(
+		Database &db,
+		uint32 character_id,
+		uint32 timestamp,
+		std::list<std::pair<int16, uint32>> &parse_list)
+	{
+		const std::string &query = fmt::format(
+			"SELECT `slot_id`, `item_id` FROM `inventory_snapshots` "
+			"WHERE `character_id` = '{}' AND `time_index` = '{}' ORDER BY `slot_id`",
+			character_id,
+			timestamp
+		);
+		auto results = db.QueryDatabase(query);
+
+		if (!results.Success()) {
+			return;
+		}
+
+		for (auto row: results) {
+			parse_list.emplace_back(std::pair<int16, uint32>(Strings::ToInt(row[0]), Strings::ToUnsignedInt(row[1])));
+		}
+	}
+
+	static void DivergeCharacterInvSnapshotFromInventory(
+		Database &db,
+		uint32 character_id,
+		uint32 timestamp,
+		std::list<std::pair<int16, uint32>> &compare_list)
+	{
+		const std::string &query = fmt::format(
+			"SELECT slot_id, item_id FROM `inventory_snapshots` "
+			"WHERE `time_index` = '{0}' AND `character_id` = '{1}' AND `slot_id` NOT IN ("
+			"SELECT a.`slot_id` FROM `inventory_snapshots` a JOIN `inventory` b USING (`slot_id`, `item_id`) "
+			"WHERE a.`time_index` = '{0}' AND a.`character_id` = '{1}' AND b.`character_id` = '{1}')",
+			timestamp,
+			character_id
+		);
+		auto results = db.QueryDatabase(query);
+
+		if (!results.Success()) {
+			return;
+		}
+
+		for (auto row: results) {
+			compare_list.emplace_back(std::pair<int16, uint32>(Strings::ToInt(row[0]), Strings::ToUnsignedInt(row[1])));
+		}
+	}
+
+	static void DivergeCharacterInventoryFromInvSnapshot(
+		Database &db, uint32 character_id, uint32 timestamp, std::list<std::pair<int16, uint32>> &compare_list)
+	{
+		const std::string &query = fmt::format(
+			"SELECT `slot_id`, `item_id` FROM `inventory` WHERE "
+			"`character_id` = '{0}' AND `slot_id` NOT IN ("
+			"SELECT a.`slot_id` FROM `inventory` a JOIN `inventory_snapshots` b USING (`slot_id`, `item_id`) "
+			"WHERE b.`time_index` = '{1}' AND b.`character_id` = '{0}' AND a.`character_id` = '{0}')",
+			character_id,
+			timestamp
+		);
+
+		auto results = db.QueryDatabase(query);
+		if (!results.Success()) {
+			return;
+		}
+
+		for (auto row: results) {
+			compare_list.emplace_back(std::pair<int16, uint32>(Strings::ToInt(row[0]), Strings::ToUnsignedInt(row[1])));
+		}
+	}
+
+	static bool SaveCharacterInvSnapshot(Database &db, uint32 character_id)
+	{
+		uint32                          time_index = time(nullptr);
+		std::vector<InventorySnapshots> queue{};
+
+		auto inventory = InventoryRepository::GetWhere(db, fmt::format("`character_id` = '{}'", character_id));
```

Comment:
Numbers shouldn't be quoted in queries, it throws off the query planner

Reply:
Updated